src/buffer.c

```diff
@@ -22,6 +22,12 @@ char git_buf__oom[1];
 	    ((d) > (b)->asize && git_buf_grow((b), (d)) < 0))\
 		return -1;
 
+int git_buffer_global_init(void)
+{
+	/* Seed rand for git_buf_put_rand. */
+	srand((unsigned int)time(NULL));
```

Comment:
We really shouldn't be using `rand()` and `srand()` at all. Seeding via `time(NULL)` only gives you a negligible amount of bits, as you can typically deduce at what time a certain command is running. I don't know about Windows, but on Unix-like systems most nowadays support calls like `getrandom()` that are much saner to use than the combination of `rand()` and `srand()`.